metadata-ingestion/src/datahub/utilities/file_backed_collections.py

```diff
@@ -270,17 +292,14 @@ def sql_query(
         return cursor.fetchall()
 
     def close(self) -> None:
-        if self._conn:
-            # Ensure everything is written out.
-            self.flush()
+        # Ensure everything is written out.
+        self.flush()
 
-            # Make sure that we don't try to use the connection anymore
-            # and that we don't drop the connection twice.
-            _sqlite_connection_cache.drop_connection(self.filename)
-            self._conn = None  # type: ignore
+        if not self.connection:  # Connection created inside this class
+            self._conn.close()
```

Comment:
if they manually call close, and then the gc calls `__del__`, we'll end up trying to delete twice?

Reply:
Yeah, I was aiming for an idempotent close. It's a little awkward because `flush` gets called multiple times (which is why I ended up adding the check on `_prune_cache`) so i'm not opposed to preventing close from getting called multiple times. However, close does end up getting called multiple times in all the tests and it ends up fine, so I don't think it's required

Reply:
oh i see - because the cleanup() call is idempotent, it should be fine